Snabble/Core/API/SnabbleAPI.swift

```diff
@@ -334,6 +330,10 @@ extension SnabbleAPI {
     // MARK: - client id
     private static let idKey = "Snabble.api.clientId"
 
+    /** SnabbleSDK client identification
+     *
+     * Stored in the keychain. Survives an uninstallation
```

Comment:
Das die Keychain die Deinstallation überlebt ist ein undokumentiertes Implementationsdetail das wir nicht garantieren können. Wenn man's dokumentiert würde ich das explizit dazusagen.

Reply:
https://developer.apple.com/documentation/security/keychain_services/keychains#

_In iOS, apps have access to a single keychain (which logically encompasses the iCloud keychain). This keychain is automatically unlocked when the user unlocks the device and then locked when the device is locked. An app can access only its own keychain items, or those shared with a group to which the app belongs. It can't manage the keychain container itself._

Ist dies nicht die Erklärung warum die Keychain ein Uninstall überlebt?

1. Single Keychain
2. Items which belong to the app or it's group

Beide Punkte sagen für mich aus das es ein Deinstallation überlebt. Wieso bist du der Meinung es ist undokumentiert?

Reply:
Für mich wäre es dann dokumentiert, wenn es explizit da steht. In der Beta von iOS 10.3 hat Apple das Verhalten tatsächlich auch geändert und ist erst zurückgerudert weil es zu viele Apps gab die sich 'drauf verlassen haben. Siehe zB auch https://developer.apple.com/forums/thread/36442 in dem "timeline"-Post von Quinn. `However, I don’t think it would surprise anyone if the 10.3 behaviour returned at some point in the future.`

Reply:
d1cfd29357da85dc5642e8d40137fffb99a1faa0